app/assets/javascripts/discourse/app/instance-initializers/banner.js

```diff
@@ -1,23 +1,25 @@
 import EmberObject from "@ember/object";
+import { setOwner } from "@ember/owner";
+import { service } from "@ember/service";
 import PreloadStore from "discourse/lib/preload-store";
 import { bind } from "discourse-common/utils/decorators";
 
-export default {
-  after: "message-bus",
+class BannerInit {
```

Comment:
I think it would be interesting to write a test for this `BannerInit` class, it's important for dev-xp to define the testing patterns so people have something to follow when they want to implement something similar or just extend the existing class. No need to write every ssing tests, but at least a foundational example would be great.

Reply:
I don't think we should attempt to unit-test initializers - they're too tightly-coupled to the rest of the application.

So, acceptance/rspec tests are gonna be best. Many of them are already covered - e.g. for this banner initializer, any issues would be caught by [this acceptance test](https://github.com/discourse/discourse/blob/04eb59a036fe69d1ad1f480179aa00bd1466e9d8/app/assets/javascripts/discourse/tests/acceptance/banner-test.js)

Reply:
Yes Im fine with acceptance, but maybe we could put them in some folder in acceptance? One problem I often have with tests is to know if it's been tested or not. In this case grepping for banner might be easy, but other cases are more tricky.

Reply:
You're thinking a folder like `tests/acceptance/initializers`?

Tricky, because the acceptance test covers the whole feature (including the initializer, components, etc.). So I'm not sure putting it in an `initializers/` folder makes sense here 🤔 

Definitely open to ideas on how we can improve this though - I agree it would be nice to have an easy way to jump from the code to the relevant tests.